Assets/Mirror/Migration/Editor/Utils.cs

```diff
@@ -69,5 +112,23 @@ public class Utils : MonoBehaviour {
                 targetProperty.SetValue(destination, srcProp.GetValue(source, null), null);
             }
         }
+
+        public static Encoding GetEncoding(string filename) {
+            // Read the BOM
+            byte[] bom = new byte[4];
+            using (FileStream file = new FileStream(filename, FileMode.Open, FileAccess.Read))
+            {
+                file.Read(bom, 0, 4);
+            }
+
+            // Analyze the BOM https://en.wikipedia.org/wiki/Byte_order_mark#Byte_order_marks_by_encoding
```

Comment:
Check first 4 bytes for known BOMs, that should help with UTF-8-BOM which is pretty common